bfvmm/src/exit_handler/src/exit_handler_dispatch.cpp

```diff
@@ -374,7 +365,24 @@ exit_handler_dispatch::handle_rsm()
 
 void
 exit_handler_dispatch::handle_vmcall()
-{ unimplemented_handler(); }
+{
+    switch (g_guest_rax)
+    {
+        case VMCS_PROMOTION:
+        {
+            advance_rip();
+            g_guest_rax = 0x00;
+            g_vcm->promote_vcpu(0);
+            break;
+        }
+        default:
+        {
+            g_guest_rax = 0xFFFFFFFFFFFFFFFF;
```

Comment:
At some point I think it would be good to have a class that handles vmcalls, so different modules could register hooks for different actions, but that's a discussion for another day.